src/presenter/trip-presenter.js

```diff
@@ -1,21 +1,79 @@
 import SortView from '../view/sort-view.js';
 import EventListView from '../view/event-list-view.js';
-import {render} from '../render.js';
+import FormEditView from '../view/form-edit-view.js';
+import NoEventView from '../view/no-event-view.js';
+import TripInfoView from '../view/trip-info-view.js';
+import { render, RenderPosition } from '../render.js';
+import { isEscapeKey } from '../util.js';
 
+const tripMainContainer = document.querySelector('.trip-main');
 export default class TripPresenter {
-  componentList = new EventListView();
+  #container = null;
+  #eventModel = null;
+
+  #componentList = new EventListView();
+
+  #eventList = [];
+
+  constructor(container, eventModel) {
+    this.#container = container;
+    this.#eventModel = eventModel;
+  }
+
+  init = () => {
+    this.#eventList = [...this.#eventModel.points];
+    this.#renderEvents();
+  };
+
+  #renderEvent = (item) => {
+    const eventComponent = new EventView(item);
+    const formEditComponent = new FormEditView(item);
 
-  init = (container, eventModel) => {
-    this.container = container;
-    this.eventModel = eventModel;
-    this.eventList = [...eventModel.getPoints()];
+    const replaceEventToForm = () => {
+      this.#componentList.element.replaceChild(formEditComponent.element, eventComponent.element);
+    };
+
+    const replaceFormToEvent = () => {
+      this.#componentList.element.replaceChild(eventComponent.element, formEditComponent.element);
+    };
+
+    const onEscKeyDown = (evt) => {
+      if (isEscapeKey) {
+        evt.preventDefault();
+        replaceFormToEvent();
+        document.removeEventListener('keydown', onEscKeyDown);
+      }
+    };
+
+    eventComponent.element.querySelector('.event__rollup-btn').addEventListener('click', () => {
+      replaceEventToForm();
+      document.addEventListener('keydown', onEscKeyDown);
+    });
+
+    formEditComponent.element.querySelector('.event__rollup-btn').addEventListener('click', () => {
+      replaceFormToEvent();
+      document.removeEventListener('keydown', onEscKeyDown);
+    });
+
+    formEditComponent.element.querySelector('form').addEventListener('submit', (evt) => {
+      evt.preventDefault();
+      replaceFormToEvent();
+      document.removeEventListener('keydown', onEscKeyDown);
+    });
+
+    render(eventComponent, this.#componentList.element);
+  };
 
-    render(new SortView(), this.container);
-    render(this.componentList, this.container);
-    render(new FormEditView(this.eventList[0]), this.componentList.getElement());
+  #renderEvents = () => {
+    if (!this.#eventList.length) {
+      render(new NoEventView(), this.#container);
+      return;
+    }
 
-    Array.from(this.eventList, (item) => render(new EventView(item), this.componentList.getElement()));
+    render(new SortView(), this.#container);
+    render(this.#componentList, this.#container);
+    Array.from(this.#eventList, (item) => this.#renderEvent(item));
```

Comment:
В целом можно использовать и такую запись, но более привычная это с использованием forEach. Меня немного смущает использование Array.from именно для модификации. Но можно и оставить
```suggestion
    this.#eventList.forEach(this.#renderEvent);
```